arch/arm/core/aarch32/mpu/arm_mpu.c

```diff
@@ -342,15 +343,14 @@ int z_arm_mpu_init(void)
 
 	arm_core_mpu_disable();
 
-#if defined(CONFIG_NOCACHE_MEMORY)
+#if defined(CONFIG_NOCACHE_MEMORY) && \
+	!defined(CONFIG_INIT_ARCH_HW_AT_BOOT)
 	/* Clean and invalidate data cache if it is enabled and
 	 * that was not already done at boot
 	 */
-#if !defined(CONFIG_INIT_ARCH_HW_AT_BOOT)
-	if (SCB->CCR & SCB_CCR_DC_Msk) {
-		SCB_CleanInvalidateDCache();
+	if (sys_cache_data_all(K_CACHE_INVD)) {
+		return -ENOTSUP;
```

Comment:
Do we really need to abort if the platform does not support a `dcache`?

Reply:
My understanding was that it is best to return any errors and also that if NOCACHE_MEMORY is on then the platform was supposed to have a DCACHE, so if this function gets called and there is no cache then potentially something is wrong, which is why I assume it has the ability to return that error code in the first place

Reply:
the original code was not returning an error so this is a change in behavior that could be unexpected for the user. I would just avoid returning here as the original code.